name.abuchen.portfolio/src/name/abuchen/portfolio/model/AttributeType.java

```diff
@@ -25,8 +27,25 @@
 
 public class AttributeType implements Named
 {
-    private static final Pattern PATTERN = Pattern.compile("^([\\d.,-]*)$"); //$NON-NLS-1$
-    private static final Pattern LIMIT_PRICE_PATTERN = Pattern.compile("^\\s*(<=?|>=?)\\s*([0-9,.']+)$"); //$NON-NLS-1$
+    private static Pattern PATTERN;
+    private static Pattern LIMIT_PRICE_PATTERN;
+
+    static
+    {
+        initPatterns();
+    }
+
+    @SuppressWarnings("nls")
+    static void initPatterns()
+    {
+        DecimalFormatSymbols dfs = DecimalFormatSymbols.getInstance(Locale.getDefault(Locale.Category.FORMAT));
+
+        // e.g. \\-?(\\d+\\.)*\\d+(\\,\\d+)?
+        String numberRegex = "\\" + dfs.getMinusSign() + "?(\\d+\\" + dfs.getGroupingSeparator() + ")*\\d+(\\"
+                        + dfs.getDecimalSeparator() + "\\d+)?";
+        PATTERN = Pattern.compile("^\\s*" + numberRegex + "\\s*$");
+        LIMIT_PRICE_PATTERN = Pattern.compile("^\\s*(<=?|>=?)\\s*(" + numberRegex + ")\\s*$");
+    }
```

Comment:
Let me try to understand the motivation for this change.

The current code first applies a pattern to check if it is kind of a normal number (and to find the indicator '>', '<', etc. for the limit price) but then parses the value the a DecimalFormat build from the current locale. If either of those fail, then an error is thrown.

To be honest, I can't remember why the code is the way it is. Because one could argue that the DecimalFormat is enough - if it parses the number, then it is a number (maybe check if there are no remaining characters left in the string).

Therefore, instead of making the pattern more complex, I wonder if we either get rid of the pattern at all *or* just add the grouping separator - say ```' ``` and a blank ``` ````. Is your concern that there are unknown separators for obscure locales?

```
 private static final Pattern PATTERN = Pattern.compile("^([\\d.,-' ]*)$"); //$NON-NLS-1$
```

Reply:
My theory about this particular logic is that checking the value to be parsed against a regex makes sure that you don't successfully parse values as correct while they aren't correct, e.g. an english formatted 1,234.56 with a german Locale. Have to check but I think the result would be 1.23456. Fun fact: The original regex wouldn't catch this problem ;-)

Of course, if it can be removed alltogether, I can remove it but my motivation was to fix the issue without changing any preexisting logic.

Reply:
> Therefore, instead of making the pattern more complex, I wonder if we either get rid of the pattern at all or just add the grouping separator - say ' and a blank  . Is your concern that there are unknown separators for obscure locales?

I'd say: Do a correct check (requiring the more complex regex I've created) or leave it away. But that's no my decision to make ;-)

Reply:
> To be honest, I can't remember why the code is the way it is. Because one could argue that the DecimalFormat is enough
After further thinking about this: DecimalFormat would also successfully parse formats using the exp-syntax like 1.2345E2. The check against the regex prevents that from happening. Maybe that was the motivation "back then" and might be a reason in favor of keeping the check today.

Reply:
The more I think about it, maybe we should use the StringToCurrencyConverter here as well. That would make the parsing consistent to other input fields.